src/sb_pointcloud_processing/src/DBSCAN.cpp

```diff
@@ -98,10 +109,92 @@ void DBSCAN::findNeighbors() {
         }
 
         // store the list of neighbours
-        this->_neighbors.insert({i, neighbors});
+        this->_neighbors[i] = neighbors;
+    }
+}
+
+void DBSCAN::findNeighborsParallel() {
+    // array to store this->num_threads number of threads
+    auto* threads = new pthread_t[this->_num_threads];
+
+    // calculate how many points in the point cloud each thread should process
+    unsigned int points_per_thread =
+    (this->_pcl.size() + this->_num_threads - 1) / this->_num_threads;
+
+    // create threads
+    for (unsigned int i = 0; i < this->_num_threads; i++) {
+        // create argument to pass in to the thread worker
+        findNeighborsThreadArg* arg =
+        buildFindNeighborsThreadArg(i, points_per_thread);
+
+        if (pthread_create(threads + i, NULL, findNeighborsThread, arg)) {
+            std::cerr << "failed to start worker" << std::endl;
+        }
     }
+
+    // join threads
+    for (unsigned int i = 0; i < this->_num_threads; i++) {
+        if (pthread_join(threads[i], NULL)) {
+            std::cerr << "failed to join worker" << std::endl;
+        }
+    }
```

Comment:
It's not quite clear how we don't just exit here? Maybe make a note about what the threads are blocking on?

Reply:
using openmp instead, not relevant anymore